File: src/main/java/com/big/dreamer/doccentral/document/carsale/model/LegalAgentDetails.java
package com.big.dreamer.doccentral.document.carsale.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;

public record LegalAgentDetails(
        @JsonProperty("nombre") @NotBlank String givenName,
        @JsonProperty("apellido") @NotBlank String lastName,
        @JsonProperty("departamento") @NotBlank String state,
        @JsonProperty("domicilio") @NotBlank String settlement,
        @JsonProperty("genero") @NotBlank String gender) {
}

File: src/main/kotlin/com/big/dreamer/doccentral/car/sale/dto/LegalAgentDetails.kt
package com.big.dreamer.doccentral.car.sale.dto

import com.fasterxml.jackson.annotation.JsonAlias

data class LegalAgentDetails (
    @JsonAlias("nombre")
    val givenName: String = "",
    @JsonAlias("apellido")
    val lastName: String = "",
    @JsonAlias("departamento")
    val state: String = "",
    @JsonAlias("domicilio")
    val settlement: String = "",
    @JsonAlias("genero")
    var gender: String = ""
)